### PR TITLE
Log additional test metrics with the CometCallback.

### PR DESCRIPTION
**Pull request by Lothiraldan**
# What does this PR do?

This PR updates the CometCallback to log metrics when checking performance using the test dataset.
It also updates the CometCallback to use the same metric naming convention as WandbCallback and MLflowCallback.

I have been working with @alexkuzmik on this PR internally so he might answer any questions or feedback directly.

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/integrations/integration_utils.py [hunk 1]
```diff
@@ -1125,6 +1126,16 @@ def on_train_end(self, args, state, control, **kwargs):
...
 
+    def on_predict(self, args, state, control, metrics, **kwargs):
+        if not self._initialized:
+            self.setup(args, state, model=None)
+        if state.is_world_process_zero:
+            if self._experiment is not None:
+                rewritten_metrics = rewrite_logs(metrics)
+                self._experiment.__internal_api__log_metrics__(
+                    rewritten_metrics, step=state.global_step, epoch=state.epoch, framework="transformers"
+                )
```
Comment (amyeroberts) [inline src/transformers/integrations/integration_utils.py:1137]: nit 

```suggestion
        if state.is_world_process_zero and if self._experiment is not None:
            rewritten_metrics = rewrite_logs(metrics)
            self._experiment.__internal_api__log_metrics__(
                rewritten_metrics, step=state.global_step, epoch=state.epoch, framework="transformers"
            )
```

**Review by amyeroberts — APPROVED** [review]
Thanks!

File: src/transformers/integrations/integration_utils.py (hunk 1 above)
Comment (alexkuzmik) [reply src/transformers/integrations/integration_utils.py]: Fixed in the latest commit.
There is an extra `if` in your suggestion :)

**Comment by amyeroberts** [comment]
Thanks for updating @Lothiraldan! 

The failing tests were because an issue we had with our test runners which should now be resolved. Could you push an empty commit to trigger a re-run? This should make everything green and mergeable :)

**Comment by Lothiraldan** [comment]
@amyeroberts I deleted our fork by mistake, do you have the possibility to re-open this PR or should I open another one?

**Comment by LysandreJik** [comment]
Hey @Lothiraldan, we unfortunately don't have the option to reopen the PR anymore

**Comment by Lothiraldan** [comment]
@LysandreJik @amyeroberts Ok I've opened this new PR https://github.com/huggingface/transformers/pull/33124, sorry about the noise again :bow:

**Pull request closed without merging**